src/main/java/org/prebid/server/bidder/yieldlab/YieldlabBidder.java

```diff
@@ -231,6 +245,63 @@ private static String getConsentParameter(User user) {
         return ObjectUtils.defaultIfNull(consent, "");
     }
 
+    private static Map<String, String> extractDsaRequestParamsFromBidRequest(BidRequest request) {
+        return Optional.ofNullable(request.getRegs())
+            .map(Regs::getExt)
+            .map(ExtRegs::getDsa)
+            .map(YieldlabBidder::extractDsaRequestParamsFromDsaRegsExtension)
+            .orElse(Map.of());
+    }
+
+    private static Map<String, String> extractDsaRequestParamsFromDsaRegsExtension(final ExtRegsDsa dsa) {
+        final HashMap<String, String> dsaRequestParams = new HashMap<>();
+
+        if (dsa.getDsaRequired() != null) {
+            dsaRequestParams.put("dsarequired", dsa.getDsaRequired().toString());
+        }
+
+        if (dsa.getPubRender() != null) {
+            dsaRequestParams.put("dsapubrender", dsa.getPubRender().toString());
+        }
+
+        if (dsa.getDataToPub() != null) {
+            dsaRequestParams.put("dsadatatopub", dsa.getDataToPub().toString());
+        }
+
+        final List<ExtRegsDsaTransparency> dsaTransparency = dsa.getTransparency();
+        if (dsaTransparency != null && !dsaTransparency.isEmpty()) {
+            dsaRequestParams.put("dsatransparency", encodeTransparenciesAsString(dsaTransparency));
+        }
+
+        return dsaRequestParams.entrySet().stream()
+            .filter(entry -> !entry.getValue().isBlank())
+            .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue));
+    }
+
+    private static String encodeTransparenciesAsString(List<ExtRegsDsaTransparency> transparencies) {
+        return transparencies.stream()
+            .filter(transparencyIsValid())
+            .map(YieldlabBidder::encodeTransparency)
+            .collect(Collectors.joining(TRANSPARENCY_TEMPLATE_DELIMITER));
+    }
+
+    private static Predicate<ExtRegsDsaTransparency> transparencyIsValid() {
```

Comment:
This should rather be the predicate itself, e.g. accept the transparency object as a parameter and return a boolean. It should be referenced as `YieldlabBidder::transparencyIsValid`. I would also call the method `isTransparencyValid`.